libs/entitlement/qa/common/src/main/java/org/elasticsearch/entitlement/qa/common/NetworkAccessCheckActions.java

```diff
@@ -53,10 +62,145 @@ static void socketBind() throws IOException {
         }
     }
 
-    @SuppressForbidden(reason = "Testing entitlement check on forbidden action")
     static void socketConnect() throws IOException {
         try (Socket socket = new DummyImplementations.DummySocket()) {
             socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
         }
     }
+
+    static void serverSocketChannelBind() throws IOException {
+        try (var serverSocketChannel = ServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+        }
+    }
+
+    static void serverSocketChannelBindWithBacklog() throws IOException {
+        try (var serverSocketChannel = ServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 50);
+        }
+    }
+
+    static void serverSocketChannelAccept() throws IOException {
+        try (var serverSocketChannel = ServerSocketChannel.open()) {
+            serverSocketChannel.bind(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0));
+            serverSocketChannel.configureBlocking(false);
+            serverSocketChannel.accept();
```

Comment:
If using bind above, you won't be testing accept in the denied case, right? Despite it being "wrong" in terms of using channels, this should just be calling `accept`